src/lsquic/client.cr

```diff
@@ -213,17 +213,24 @@ module QUIC
         end
         @engine_open = false
         LibLsquic.engine_destroy(engine)
+        @socket.try &.close
+        @socket = nil
```

Comment:
@vhuynh3000 why try to close the socket in this block?

Reply:
The socket is no longer used after the connection is closed and the engine is destroyed. However there is a separate fiber continues to run since the socket.read call has no timeout. This leaves an open socket every time the connection is closed. By closing the socket here, it raises an exception in other fiber (since the read can't occur when the socket closed), and allows the fiber to complete.